utils/das_data_sorter.py

```diff
@@ -1,39 +1,43 @@
 import pandas as pd 
 from argparse import ArgumentParser
-from numpy import ceil
+from numpy import ceil, median
 
 # accepts terminal arguments
 parser = ArgumentParser()
-parser.add_argument("--input", help="Reads the inputted CSV file to filter", action="store", required=True)
-parser.add_argument("--output", help="Writes the filtered data onto a new CSV file under this name", action="store", required=True)
-parser.add_argument("--unit", help="Specify time units [seconds, s, or minutes, m]. Default is in seconds.", default="seconds", 
+parser.add_argument("-i", "--input", help="Reads the inputted CSV file to filter", action="store", required=True)
+parser.add_argument("-o", "--output", help="Writes the filtered data onto a new CSV file under this name", action="store", required=True)
+parser.add_argument("--unit", help="Specifies time units [seconds, s, or minutes, m]. Default is in seconds.", default="seconds", 
                     choices=["seconds", "s", "minutes", "m"], action="store")
+parser.add_argument("--smooth", help="Smooths data points using N-point mean or median smoothing", choices=["mean", "median"], action="store")
+parser.add_argument("-n", help="Specifies number of data points taken for smoothing", default=3, action="store", type=int)
 args = parser.parse_args()
 
 class DasSort:
     def __init__(self, file_input:pd.DataFrame, unit:str) -> None:
         self.indexes = None
-        self.time = self.convert_time(file_input["time"], unit)
-        self.gps = self.gps_data(file_input["gps"])
-        self.gps_lat = self.average_data(file_input["gps_lat"])
-        self.gps_long = self.average_data(file_input["gps_long"])
-        self.gps_alt = self.average_data(file_input["gps_long"])
-        self.gps_course = self.average_data(file_input["gps_course"])
-        self.gps_speed = self.average_data(file_input["gps_speed"])
-        self.gps_satellites = self.average_data(file_input["gps_satellites"])
-        self.ax = self.average_data(file_input["aX"])
-        self.ay = self.average_data(file_input["aY"])
-        self.az = self.average_data(file_input["aZ"])
-        self.gx = self.average_data(file_input["gX"])
-        self.gy = self.average_data(file_input["gY"])
-        self.gz = self.average_data(file_input["gZ"])
-        self.thermoc = self.average_data(file_input["thermoC"])
-        self.thermof = self.average_data(file_input["thermoF"])
-        self.pot = self.average_data(file_input["pot"])
-        self.cadence = self.average_data(file_input["cadence"])
-        self.power = self.average_data(file_input["power"])
-        self.reed_velocity = self.average_data(file_input["reed_velocity"])
-        self.reed_distance = self.average_data(file_input["reed_distance"])
+        self.data = {
```

Comment:
This is definitely a good candidate to refactor later on! But no need to worry about it for now!